src/mscorlib/shared/System/Buffers/Utilities.cs

```diff
@@ -14,6 +14,9 @@ internal static int SelectBucketIndex(int bufferSize)
         {
             Debug.Assert(bufferSize >= 0);
 
+            // bufferSize of 0 will overflow here, causing a huge
+            // index which the caller will discard because it is not
+            // within the bounds of the bucket array
```

Comment:
Nit: maybe say "will underflow" rather than "will overflow", and add a period at the end of the sentence.